controller/Review.php

```diff
@@ -118,6 +138,11 @@ public function index(): void
         } else {
             $execution = $finder->findDeliveryExecution($launchData);
         }
+
+        if ($this->isPausingConcurrentSessionsEnabled()) {
+            $this->pauseConcurrentSessions($execution);
+        }
```

Comment:
I would also move the decision of what to do to the new service (locate on other extension most probably delivery extension).

```suggestion
            $this->pauseService->pauseConcurrentSessions($execution);
```